cores/esp8266/WString.h

```diff
@@ -250,27 +250,29 @@ class String {
             uint16_t cap;
             uint16_t len;
         };
-
-        // SSO is handled by checking the last byte of sso_buff.
-        // When not in SSO mode, that byte is set to 0xff, while when in SSO mode it is always 0x00 (so it can serve as the string terminator as well as a flag)
-        // This allows strings up up to 12 (11 + \0 termination) without any extra space.
-        enum { SSOSIZE = sizeof(struct _ptr) + 4 }; // Characters to allocate space for SSO, must be 12 or more
+        // This allows strings up up to 11 (10 + \0 termination) without any extra space.
+        enum { SSOSIZE = sizeof(struct _ptr) + 4 - 1 }; // Characters to allocate space for SSO, must be 12 or more
+        struct _sso {
+            char     buff[SSOSIZE];
+            unsigned len  : 7;
```

Comment:
Shouldn't this be unsigned char?

Reply:
Talking about len or buff?  `buff` should be `char` as that's the non-SSO type.  `len` is a bitfield so `unsigned` or `signed` will do the trick.

Reply:
I mean unsigned char (1 byte)  as opposed to just unsigned (4 bytes)